ledger/complete/checkpoint/trigger.go

```diff
@@ -0,0 +1,112 @@
+package checkpoint
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/onflow/flow-go/ledger/complete/mtrie/trie"
+	"github.com/rs/zerolog"
+	"go.uber.org/atomic"
+)
+
+type runner interface {
+	runCheckpoint(ctx context.Context, checkpointTries []*trie.MTrie, checkpointNum int) error
+}
+
+type TrieQueue interface {
+	Tries() []*trie.MTrie
+}
+
+type Trigger struct {
+	logger                  zerolog.Logger
+	activeSegmentNum        int                     // the num of the segment that the previous trie update was written to, is used to detect whether trie update is written to a new segment.
+	checkpointedSegumentNum atomic.Int32            // the highest segment num that has been checkpointed
```

Comment:
This is one change I made. 

The previous solution is to save the next segment num in the future to trigger checkpointing. And my change is to save the last checkpointed segment num.

The difference is that we don't need to update this value when checkpointing retry is needed.